tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -0,0 +1,321 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Romayne (https://github.com/MeggalBozale)
+ * @license ISC
+ */
+
+import { useBackend, useLocalState, useSharedState } from '../../backend';
+import { Window } from '../../layouts';
+import { toTitleCase } from 'common/string';
+import { Button, Collapsible, Divider, Input, LabeledList, ProgressBar, Section, Slider, Stack } from '../../components';
+import { formatMoney } from '../../format';
+import { MaintenencePanel, Manufacturable, ManufacturerData, Ore, QueueBlueprint, Resource, Rockbox } from './type';
+
+import { BlueprintButton } from './blueprintButton';
+import { ProductionCard } from './productionCard';
+import { clamp } from 'common/math';
+import { CollapsibleWireMenu } from './collapsibleWireMenu';
+import { pluralize } from '../common/stringUtils';
+
+const CardInfo = (_, context) => {
+  const { data, act } = useBackend<ManufacturerData>(context);
+  return (data.card_owner === null || data.card_balance === null) ? (
+    <Section
+      textAlign="center"
+    >
+      <Stack vertical>
+        <Stack.Item>
+          No Account Found.
+        </Stack.Item>
+        <Stack.Item>
+          <Button icon="add" onClick={() => act("card", { "scan": true })}>Add Account</Button>
+        </Stack.Item>
+      </Stack>
+    </Section>
+  ) : (
+    <Section
+      title="Account Info"
+      buttons={<Button icon="minus" onClick={() => act("card", { "remove": true })}>Log Out</Button>}
+    >
+      <LabeledList>
+        <LabeledList.Item
+          label="Owner"
+        >
+          {data.card_owner}
+        </LabeledList.Item>
+        <LabeledList.Item
+          label="Balance"
+        >
+          {formatMoney(data.card_balance)}⪽
+        </LabeledList.Item>
+      </LabeledList>
+    </Section>
+  );
+};
+
+export const Manufacturer = (_, context) => {
+  const { act, data } = useBackend<ManufacturerData>(context);
+  const [repeat, toggleRepeatVar] = useSharedState(context, "repeat", data.repeat);
+  const [search, setSearchData] = useLocalState(context, "query", "");
+  const [swap, setSwappingMaterial] = useLocalState(context, "swap", null);
+  const wirePanel:MaintenencePanel = {
+    indicators: data.indicators,
+    wires: data.wires,
+    wire_bitflags: data.wire_bitflags,
+  };
+  let toggleRepeat = () => {
+    act("repeat");
+    toggleRepeatVar(!repeat);
+  };
+  let swapPriority = (materialID: string) => {
+    if (swap === null) {
+      setSwappingMaterial(materialID);
+    }
+    else if (swap === materialID) {
+      setSwappingMaterial(null);
+    }
+    else {
+      act("material_swap", { "resource_1": swap, "resource_2": materialID });
+      setSwappingMaterial(null);
+    }
+  };
+  const all_blueprint_lists = [
+    data.available_blueprints,
+    data.downloaded_blueprints,
+    data.recipe_blueprints,
+    data.hidden_blueprints,
+  ];
+  const all_blueprint_list_strings = [
+    "available",
+    "download",
+    "drive_recipes",
+    "hidden",
+  ];
```

Comment:
Object with string keys for the type and values being the `data.foo_blueprints` values, surely? At minimum array of tuples of `["available", data.available_blueprints]` etc.

Reply:
dictionary also should work as im under the impression thats what you mean. might be mistaken but this seems to be significantly better